### PR TITLE
Support importing types from nested packages

### PR DESCRIPTION
**Pull request by macbeth322**
Attempts to address problems discussed in issue #1645. I enhanced the `mypy.build._find_module` to search for `py.typed` marker files at all package levels instead of at the top level. This change currently doesn't update stub package loading but I have added a test that fails without my change and passes with it.

### CODE REVIEW

**Comment by macbeth322** [comment]
Is anyone available to look at this?

**Comment by ethanhs** [comment]
Ah, sorry this PR got under my radar. I will take a look later this week.

**Comment by macbeth322** [comment]
Hi @ethanhs, were you able to have a look at this last week?

**Comment by gvanrossum** [comment]
I'll review this unless @ethanhs steals it back from me again.

File: mypy/build.py [hunk 1]
```diff
@@ -896,6 +896,23 @@ def find_module(self, id: str, search_paths: SearchPaths,
...
+        pkgs2check = (
+            fscache.isfile(
+                os.path.join(os.path.join(pkg_dir, *components[:count + 1]), 'py.typed')
+            ) for count in range(len(components))
+        )
+        if next(pkgs2check):
+            return os.path.join(pkg_dir, *components[:-1]), True
+        if any(pkgs2check):
+            return os.path.join(pkg_dir, *components[:-1]), False
+        return None
```
[the remark this answers is not in the document]
Comment (macbeth322) [reply mypy/build.py]: I rewrote this function to have the same functionality, but a little more imperatively.

File: mypy/build.py [hunk 2]
```diff
@@ -915,7 +932,6 @@ def _find_module(self, id: str, search_paths: SearchPaths,
         # Third-party stub/typed packages
```
[the remark this answers is not in the document]
Comment (macbeth322) [reply mypy/build.py]: I added type annotations to the `third_party_*_dirs` variables. If there was something else you meant by your comment, please let me know.

File: mypy/build.py [hunk 3]
```diff
@@ -2357,11 +2376,13 @@ def find_module_and_diagnose(manager: BuildManager,
             if not (options.ignore_missing_imports or in_partial_package(id, manager)):
                 module_not_found(manager, caller_line, caller_state, id)
             raise ModuleNotFound
-        else:
+        elif root_source:
```
[the remark this answers is not in the document]
Comment (macbeth322) [reply mypy/build.py]: Personally, I prefer the explicitness of using `elif` when chaining if statements but if its an issue I can change it.

[the remark this answers is not in the document]
Comment (gvanrossum) [reply mypy/build.py]: In that case it would behoove you to change the `if` on L2375 to `elif` as well. :-) I'll let it slide.

**Pull request merged**